scripts/parse_tools/parse_checkers.py

```diff
@@ -12,7 +12,7 @@
 
 ########################################################################
 
-_UNITS_RE = re.compile(r"^[^/@#$%^&*()\|<>\[\]{}?,.]+$")
+_UNITS_RE = re.compile(r"^[^/!@#$%^&*=()\|<>\[\]{}?,.]+$")
```

Comment:
Really? What do examples that use these look like?

Reply:
Since it's impossible to decipher regular expressions like this, it might be good to add a few examples above what we want to match. This would also help addressing @gold2718's question.

Reply:
After staring at this for a minute, I think we can actually clean this up a bit if it makes sense with something like:
```python
unitless_regex = "1"
unit_regex = "[a-zA-Z]+[-]?[0-9]+"
units_regex = f"^({unit_regex}(\s{unit_regex})?|{unitless_regex})$"
# >>> print(units_regex)
# ^([a-zA-Z]+[-]?[0-9]+(\s[a-zA-Z]+[-]?[0-9]+)?|1)$
_UNITS_RE = re.compile(units_regex)
```
Testing locally and on regex101 looks to allow only what we want minus the exponents with leading `0`'s (like `m s-01`) but that's easy enough to add in, but I was just curious if this was a reasonable replacement first?

Reply:
@climbfuji @nusbaume thoughts on @mwaxmonsky 's regex cleanup?

Reply:
Did a little more testing and I missed two common cases so I think a more flushed out version could be:
```python
unitless_regex                = "1"
non_leading_zero_num          = "[1-9]\d*"
negative_non_leading_zero_num = f"[-]{non_leading_zero_num}"
unit_exponent                 = f"({negative_non_leading_zero_num}|{non_leading_zero_num})"
unit_regex                    = f"[a-zA-Z]+{unit_exponent}?"
units_regex                   = f"^({unit_regex}(\s{unit_regex})*|{unitless_regex})$"
# >>> print(units_regex)
# ^([a-zA-Z]+([-][1-9]\d*|[1-9]\d*)?(\s[a-zA-Z]+([-][1-9]\d*|[1-9]\d*)?)*|1)$
_UNITS_RE = re.compile(units_regex)
```

Reply:
I am not super familiar with this particular collection of functions, but I personally think @mwaxmonsky's method of breaking up the regex into smaller chunks makes it easier to understand.  It also appears to split the target string into specific regex groups, which could potentially be useful in the future.

The one potentially important difference I have found is that the original regex would match strings with leading or trailing white spaces, while the new one does not.  Do we ever expect the target string to have leading or trailing white spaces? 

Finally, if we are worried about what this regex is doing then we can always throw a lot more unit tests at it to make sure we are covering as many edge/corner cases as we can think of (which we can always do in a future PR if that is of interest).

So to summarize, I am personally happy with the changes, but will always leave the final decision up to the core framework developers.

Reply:
I'm not as familiar with the backwards compatibility needed for the whitespace support but I think we can leave that out of the regex and pass the proposed units `str` through python's `strip()` function to remove all surrounding whitespace and then pass it to `_UNITS_RE.match(...)`.

Reply:
I like @mwaxmonsky's solution!

Reply:
Update pushed.  Just let me know if the new version has any issues!

Reply:
Also on the documentation front, I noticed that there's a few `doctest` tests which pass in the units and error on invalid strings.  Would we be duplicating documentation by putting in examples outside the doctest for `check_units()` right below the regex declaration?

Reply:
@mwaxmonsky has pushed his regex updates, which is both cleaner and easier to decipher!